src/parser/smt2/smt2_lexer_new.cpp

```diff
@@ -0,0 +1,385 @@
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * The lexer for smt2
+ */
+
+#include "parser/smt2/smt2_lexer_new.h"
+
+#include <cstdio>
+
+#include "base/check.h"
+#include "base/output.h"
+#include "parser/flex_lexer.h"
+
+namespace cvc5 {
+namespace parser {
+
+Smt2LexerNew::Smt2LexerNew(bool isStrict, bool isSygus)
+    : FlexLexer(),
+      d_peekedChar(false),
+      d_chPeeked(0),
+      d_isStrict(isStrict),
+      d_isSygus(isSygus)
+{
+  for (size_t i = 0; i < 256; i++)
+  {
+    d_symcTable[i] = CharacterClass::NONE;
+  }
+  for (char ch = 'a'; ch <= 'z'; ++ch)
+  {
+    d_symcTable[static_cast<size_t>(ch)] = CharacterClass::SYMBOL;
+  }
+  for (char ch = 'A'; ch <= 'Z'; ++ch)
+  {
+    d_symcTable[static_cast<size_t>(ch)] = CharacterClass::SYMBOL;
+  }
+  for (char ch = '0'; ch <= '9'; ++ch)
+  {
+    d_symcTable[static_cast<size_t>(ch)] = CharacterClass::DECIMAL_DIGIT;
+  }
+  // ~!@$%^&*_-+=<>.?/
+  d_symcTable[static_cast<size_t>('~')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('!')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('@')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('$')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('%')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('^')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('&')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('*')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('_')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('-')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('+')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('=')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('<')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('>')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('.')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('?')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>('/')] = CharacterClass::SYMBOL;
+  d_symcTable[static_cast<size_t>(',')] = CharacterClass::SYMBOL;
+  // whitespace
+  d_symcTable[static_cast<size_t>(' ')] = CharacterClass::WHITESPACE;
+  d_symcTable[static_cast<size_t>('\t')] = CharacterClass::WHITESPACE;
+  d_symcTable[static_cast<size_t>('\r')] = CharacterClass::WHITESPACE;
+  d_symcTable[static_cast<size_t>('\n')] = CharacterClass::WHITESPACE;
+}
+
+const char* Smt2LexerNew::tokenStr() const
+{
+  Assert(!d_token.empty() && d_token.back() == 0);
+  return d_token.data();
+}
+bool Smt2LexerNew::isStrict() const { return d_isStrict; }
+bool Smt2LexerNew::isSygus() const { return d_isSygus; }
+
+Token Smt2LexerNew::nextTokenInternal()
```

Comment:
Did you try moving this to header for inlining/speed-up? You will probably have to remove the trace tags to see any improvement.

Reply:
This one won't be a bottleneck, since it is per token, not per character.

Reply:
Yeah, was just curious, I haven't moved this one to the header in the Bitwuzla lexer, either.